### PR TITLE
python/scrabble-score: Update mentoring.md

### PR DESCRIPTION
**Pull request by feature-engineer**
1. Using a dictionary for mapping score to letters only for it to be inverted, isn't necessary, and makes it less similar to the original table. Using tuples makes it both more readable and more efficient.

2. No need to indicate global and compute the dictionary once the function gets called, just use a constant outside the function. 

3. Name constants in all upper case, as per PEP 8.

4. Use a generator in sum, not a list.

### CODE REVIEW

**Comment by SleeplessByte** [comment]
Are the changes you propose _more idiomatic_ or are they just preferences? If it's the latter, don't remove the old solutions/suggestions.

**Comment by feature-engineer** [comment]
@cmccandless If you don't use the const convention for the dictionary by using all lower case, you'd get linters complaining about not using the const convention, for example pylint would issue the following:

>  C0103: Constant name "letters_to_scores" doesn't conform to UPPER_CASE naming style (invalid-name)

Constants aren't necessarily immutable, they're just variables which you define only once, and can't overwrite. It would have been better to have an immutable const, but I think that introducing MappingProxyType is too much for this exercise. 

Was there any other change you wanted to have?

**Comment by SleeplessByte** [comment]
> Constants aren't necessarily immutable, they're just variables which you define only once, and don't want to overwrite. It would have been better to have an immutable const, but I think that introducing MappingProxyType is too much for this exercise.

In this case, you can show _intent_ by using `UPPER_CASE`.

**Comment by cmccandless** [comment]
OK, I'm fine with keeping the changes to UPPER_CASE.

To summarize my suggestions:
### KEEP
- changes to make global variables UPPER_CASE
- conversion to iterator inside `sum()`
### REVERT
- changes to dictionary structure. The example here is written in a particular way  to demonstrate certain concepts (see the text above the code example), and for that reason should not be altered significantly.

**Comment by yawpitch** [comment]
The intent of both the globals is to create a constant, so UPPER_CASE is correct in this case, though @feature-engineer it’s important to note that Pylint can often false positive on globals that are actually variable due to mutations elsewhere.

Personally, though, I’d tend to prefer a simple dict literal to this “tuple of tuples and reverse” work. There needs be _no_ runtime calculation cost to importing this module, and an appropriately sorted dict is no harder to maintain.

Also, said dict need not even include entries for any of the 1 point characters, so it’s not a significant number of lines to create.

**Comment by feature-engineer** [comment]
@yawpitch I'm not sure I understand how Enum would be used here.

**Comment by yawpitch** [comment]
For ease of `sum` it would be the IntEnum.

```
from enum import IntEnum

class Letters(IntEnum):
    A = E = I = O = U = L = N = R = S = T = 1
    D = G = 2
    B = C = M = P = 3
    F = H = V = W = Y = 4
    K = 5
    J = X = 8
    Q = Z = 10
    
def score(word: str) -> int:
     return sum(Letters[c] for c in word.upper())
```

**Comment by feature-engineer** [comment]
I like @yawpitch 's suggested solution, it's better than the dictionary inversion, or dictionary literal.

Is there any objection to replace the dictionary inversion completely with the Enum approach? I don't see any reason to keep the dictionary inversion.

File: tracks/python/exercises/scrabble-score/mentoring.md [hunk 1]
```diff
@@ -5,12 +5,12 @@ In Scrabble, rare letters are worth more than common letters.
...
 	
-The basic solution has two parts: building a dictionary called
-letter_value here, and using it. Using it is quite simple:
+The basic solution has two parts: defining a global dictionary using a 
+dictionary literal called LATTER_VALUE, and using it. Using it is quite simple:
 
 ```python
 def score(word: str) -> int:
-    return sum( [ letter_value[ch] for ch in word.upper() ] )
+    return sum( [ LATTER_VALUE[ch] for ch in word.upper() ] )
```
Comment (cmccandless) [inline tracks/python/exercises/scrabble-score/mentoring.md]:
```suggestion
    return sum( [ LETTER_VALUE[ch] for ch in word.upper() ] )
```

File: tracks/python/exercises/scrabble-score/mentoring.md [hunk 2]
```diff
@@ -5,12 +5,12 @@ In Scrabble, rare letters are worth more than common letters.
 
 ### Reasonable solutions
 	
-The basic solution has two parts: building a dictionary called
-letter_value here, and using it. Using it is quite simple:
+The basic solution has two parts: defining a global dictionary using a 
+dictionary literal called LATTER_VALUE, and using it. Using it is quite simple:
```
Comment (cmccandless) [inline tracks/python/exercises/scrabble-score/mentoring.md]:
```suggestion
dictionary literal called LETTER_VALUE, and using it. Using it is quite simple:
```

File: tracks/python/exercises/scrabble-score/mentoring.md [hunk 3]
```diff
@@ -38,64 +38,89 @@ to get a mapping from letters to their value in the game.
...
     3 : 'B, C, M, P',
     4 : 'F, H, V, W, Y',
     5 : 'K',
     8 : 'J, X',
     10: 'Q, Z'
-    }
+}
+    
+LETTERS_TO_SCORES = { 
+    ch: score for score, letters  in LETTERS_AND_SCORES
```
Comment (cmccandless) [inline tracks/python/exercises/scrabble-score/mentoring.md]: Iterating through a `dict` object returns an iterator of the keys only.

```suggestion
    ch: score for score, letters in LETTERS_AND_SCORES.items()
```

**Pull request merged**